lib/taskana-core/src/main/java/pro/taskana/impl/TaskQueryImpl.java

```diff
@@ -117,11 +181,15 @@ public ObjectReferenceQuery createObjectReferenceQuery() {
     @Override
     public List<Task> list() throws NotAuthorizedException {
         LOGGER.debug("entry to list(), this = {}", this);
-        List<Task> result = null;
+        List<Task> result = new ArrayList<>();
         try {
             taskanaEngineImpl.openConnection();
             checkAuthorization();
-            result = taskanaEngineImpl.getSqlSession().selectList(LINK_TO_MAPPER, this);
+            List<TaskImpl> tasks = taskanaEngineImpl.getSqlSession().selectList(LINK_TO_MAPPER, this);
+            for (TaskImpl taskImpl : tasks) {
```

Comment:
Can be done as lambda with J8.    result.stream().foreach(r -> setpr(r));   so it can be directly stored into result and it´s better/faster to understand.